build.c

```diff
@@ -319,6 +327,15 @@ jobstart(struct job *j, struct edge *e)
 		warn("posix_spawn_file_actions_addclose:");
 		goto err3;
 	}
+	if (isjobserverclient()) {
+		/* do not allow children to steal GNU/tokens */
```

Comment:
As @trws asked, do we need to prevent children from interacting with the job server?

If we do, I think it'd be better to set `FD_CLOEXEC` on the fds than closing them here.

Reply:
I don't remember why I thought of that in the first place (nowhere in the documentation is this recommended).

After testing for a bit, it seems GNU Make uses the same `--jobserver-auth` parameters for every client, so it's fine for `samu` to share the jobserver with its own clients.